foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/actions/GroupAction.java

```diff
@@ -156,68 +158,53 @@ public void validateImpl(GroupRequest parameter) throws MalformedQueryException
-        FieldMetadata meta = metaMap.get(field);
 
+        Interval queryInterval = new PeriodSelector(parameter.getFilters()).analyze();
+        long minutes = queryInterval.toDuration().getStandardMinutes();
+
+        String field = parameter.getNesting().get(0);
+        FieldMetadata meta = metaMap.get(field);
         if (null == meta || null == meta.getEstimationData()) {
             log.warn("No estimation data found for field {} of table {}", field, parameter.getTable());
             return 0.0;
         }
         final long count = meta.getEstimationData().getCount();
 
-        long estimatedDocs = (count * minutes) / 1440;
+        long estimatedDocs = (count * minutes) / TimeUnit.DAYS.toMinutes(1);
 
+        //Checking explicitly for the first grouping to ignore the algo
         if (estimatedDocs < MIN_ESTIMATION_THRESHOLD) { //Not worth checking
             log.debug("Estimator skipped as # docs ({}) < threshold ({})",
                     estimatedDocs, MIN_ESTIMATION_THRESHOLD);
             return 0.0;
         }
-        List<Filter> filters = parameter.getFilters();
-        log.debug("Starting estimation with: {}", meta);
-        double probability = meta.getEstimationData().accept(new EstimationDataVisitor<Double>() {
-            @Override
-            public Double visit(FixedEstimationData fixedEstimationData) {
-                final double result = (fixedEstimationData.getProbability() / 100);
-                log.debug("From fixed probability: {} / 100 = {}",
-                        fixedEstimationData.getProbability(),
-                        result);
-                return result;
-            }
 
-            @Override
-            public Double visit(PercentileEstimationData bucketBasedEstimationData) {
-                log.debug("Returning fixed estimation data 1.0");
-                return 1.0;
-            }
+        log.debug("Starting estimation with: {} estimatedDocs:{}", meta, estimatedDocs);
+        Double probability = 0.0;
 
-            @Override
-            public Double visit(CardinalityEstimationData cardinalityBasedEstimationData) {
-                final double result = (
-                        (double) (Long.min(
-                                cardinalityBasedEstimationData.getCardinality(),
-                                cardinalityBasedEstimationData.getCount()))
-                                / (double) cardinalityBasedEstimationData.getCount());
-                log.debug("Result from cardinality estimation: min({}, {})/ {} = {}",
-                        cardinalityBasedEstimationData.getCardinality(),
-                        cardinalityBasedEstimationData.getCount(),
-                        cardinalityBasedEstimationData.getCount(),
-                        result);
-                return result;
-            }
-        });
-        log.debug("First phase probability for field {} is {}", field, probability);
+        for (String innerField : parameter.getNesting()) {
```

Comment:
@santanusinha @r0goyal Checking cardinality for inner fields too
There might be a case when the nested field has very high cardinality which can bring down the cluster
Taking the maximum probability among all the nested field to be on the safe side